src/Xrefcheck/Scan.hs

```diff
@@ -89,6 +89,20 @@ specificFormatsSupport formats = \ext -> M.lookup ext formatsMap
         , extension <- extensions
         ]
 
+-- | Process files that are tracked by git and not ignored by the config.
+readDirectoryWith
+  :: TraversalConfig
+  -> (FilePath -> IO a)
+  -> FilePath
+  -> IO [(FilePath, a)]
+readDirectoryWith config scanner root =
+  traverse scanFile
+  . filter (not . isIgnored)
+  . L.lines =<< readProcess "git" ["ls-files", root] ""
```

Comment:
I am not sure this works as you expect

`~/serokell` is a folder where I clone different repositories

```
anton@RedmiBook:~/serokell$ git ls-files xrefcheck
fatal: not a git repository (or any of the parent directories): .git
anton@RedmiBook:~/serokell$ git ls-files ~/serokell/xrefcheck
fatal: not a git repository (or any of the parent directories): .git
```

Reply:
And if I do `cd xrefcheck && git ls-files`, everything is fine